Feat: Trn 48 compose django with postgre sql

jestebanpelaez18 opened the pull request:
Django and PosgreSQL are compose now, to containers are running properly. If you want to take a look you can redirect to this link in google Chrome: http://localhost:8000/ if its works it means that is okey. The next step will be test it with postman at least.

jestebanpelaez18 committed:
remove: removed juan php

jestebanpelaez18 committed:
fix: changed .dev.env file to .env, had some issues with.dev.env

jestebanpelaez18 committed:
fix: fixed syntax issues in the docker compose and set the enviroment variables in the Dockerfile

jestebanpelaez18 committed:
add: added psycopg2-bynary installation package in the requeriments.txt which is important for config PostGreSQL in django

jestebanpelaez18 committed:
add : Database section in setting.py that allow the connection with POSGRESQL

jestebanpelaez18 committed:
add : Database section in setting.py that allow the connection with POSGRESQL

jestebanpelaez18 committed:
remove: Removed py cache from all the filesç

RushMaverick reviewed (approved):
Approved! If you only changed variables, please tag the commit as <chore>! Anyhow, good stuff, moving on!

Stte commented on backend/django_backend/django_backend/settings.py:
Should we use variables from .dev.env?

jestebanpelaez18 replied:
i tried but it does not work, don't know why. I will take a look though

jestebanpelaez18 replied:
it does not work for the database when I'm using .dev.env, not sure why because when I use .env works perfect

Stte commented on docker-compose.dev.yml:
```diff
@@ -21,15 +21,15 @@ services:
     container_name: postgres_db
     ports:
-      -"5432:5432"
+      - "5432:5432"
```
We can remove ports and still connect django because they are in the same network although we should probably make that explicit by adding the network here.

jestebanpelaez18 replied:
Hey i added the new network check it please, its working fine with me !

Stte commented on docker-compose.dev.yml:
I got error on the first try that django couldn't connect to db.
I think it can be fixed like this: https://stackoverflow.com/a/55835081

jestebanpelaez18 replied:
I fix the bug with those recommendations, its working for me. Check it and if there is a problem we can created a script that make django waits until the database is ready, but I applied the solution provided in the article and work pretty good!

Stte reviewed (changes requested):
Good job! I tested that the db works by first creating a super user with `python manage.py createsuperuser` and then login in to `http://localhost:8000/admin/`. 

There are few changes that might be good to add. One would be to add clean to makefile that would delete the db volume.

jestebanpelaez18 committed:
Chore(Makefile): add new rules for managing Docker containers such as volume, net ands ps.
Modified previous rules such all, clean, fclean and all

jestebanpelaez18 committed:
fix(docker-compose.dev.yml): solve the problem that causes django start before than posgreSQL.

Added in db services healtcheck property, that will allow to check the status of the Database, and until the database is not ready Django its not gonna start,so added the condition in django service.

jestebanpelaez18 committed:
feat(docker-compose.dev.yml): introduce new network for service isolation.

Created network django_db_network which is the network that communicate the database and Django.

Stte committed:
fix(Makefile): change docker-compose to docker compose

Apparently docker-compose is older version and doesn't always work.

Stte reviewed (approved):
Hell yeah! Good job!

RushMaverick commented:
You can merge it if you are ready with the changes!
